sklearn/utils/tests/test_param_validation.py

```diff
@@ -116,12 +116,13 @@ def test_interval_errors(params, error, match):
 
 def test_stroptions():
     """Sanity check for the StrOptions constraint"""
-    options = StrOptions({"a", "b", "c"}, deprecated={"c"})
+    options = StrOptions({"a", "b", "c"}, deprecated={"c"}, internal={"b"})
```

Comment:
Naive questions: must we test that `internals` and `deprecated` are disjoint subsets of `options`? Could this prevent subsequent development mistakes?

Reply:
Makes sense. I added a check for that and a test